sound/soc/sof/trace.c

```diff
@@ -388,9 +389,12 @@ int snd_sof_init_trace_ipc(struct snd_sof_dev *sdev)
 	if (!sdev->dtrace_is_supported)
 		return 0;
 
-	if (sdev->dtrace_is_enabled || !sdev->dma_trace_pages)
+	if (sdev->dtrace_state == SOF_DTRACE_ENABLED || !sdev->dma_trace_pages)
 		return -EINVAL;
 
+	if (sdev->dtrace_state == SOF_DTRACE_STOPPED)
+		goto start;
```

Comment:
I personally avoid goto use for non-error logic, but as others have been ok with this, I'm good with as well.  The logic and the jump label seem clear in this case.

Reply:
I'm not sure how this would look with
```c
if (sdev->dtrace_state != SOF_DTRACE_STOPPED) {
}
```
I think the goto is fine